src/service/fixed-layer.js

```diff
@@ -320,28 +312,15 @@ export class FixedLayer {
             continue;
           }
 
-          // Calculate top, assuming that it could implicitly be `auto`.
-          // `getComputedStyle().top` will return `auto` in Safari and the
-          // actual calculated value in all other browsers. To find out whether
-          // or not the `top` was actually set in CSS, this method compares
-          // `offsetTop` with `style.top = 'auto'` and without.
-          let top = style.top;
-          const currentOffsetTop = element./*OK*/offsetTop;
-          if (isSticky) {
-            if (top === 'auto' || parseInt(top, 10) !== autoTops[i]) {
-              top = '';
-            }
-          } else if (currentOffsetTop === autoTops[i]) {
-            if (currentOffsetTop ===
-                    this.committedPaddingTop_ + this.borderTop_) {
+          if (top === 'auto' || autoTops[i] !== top) {
+            if (isFixed &&
+                offsetTop === this.committedPaddingTop_ + this.borderTop_) {
               top = '0px';
             } else {
               top = '';
             }
           }
 
-          const bottom = style.bottom;
-          const opacity = parseFloat(style.opacity);
```

Comment:
Typically we try to stay away from numerical comparisons with implicit cast. Are we scrapping that?